engine/render/src/render/render_script.cpp

```diff
@@ -2213,26 +2248,22 @@ namespace dmRender
 
         RenderScriptInstance* i = RenderScriptInstance_Check(L);
         luaL_checktype(L, 1, LUA_TTABLE);
-        if (i->m_PredicateCount < MAX_PREDICATE_COUNT)
-        {
-            dmRender::Predicate* predicate = new dmRender::Predicate();
-            i->m_Predicates[i->m_PredicateCount++] = predicate;
-            lua_pushnil(L);  /* first key */
-            while (lua_next(L, 1) != 0)
-            {
-                predicate->m_Tags[predicate->m_TagCount++] = dmScript::CheckHashOrString(L, -1);
-                lua_pop(L, 1);
-                if (predicate->m_TagCount == dmRender::Predicate::MAX_TAG_COUNT)
-                    break;
-            }
-            lua_pushlightuserdata(L, (void*)predicate);
-            assert(top + 1 == lua_gettop(L));
-            return 1;
-        }
-        else
+
+        HPredicate* p_predicate = (HPredicate*) lua_newuserdata(L, sizeof(HPredicate*));
+        *p_predicate = NewPredicate();
+
+        luaL_getmetatable(L, RENDER_SCRIPT_PREDICATE);
+        lua_setmetatable(L, -2);
+
+        lua_pushnil(L);  /* first key */
+        while (lua_next(L, 1) != 0)
         {
-            return luaL_error(L, "Could not create more predicates since the buffer is full (%d).", MAX_PREDICATE_COUNT);
+            dmhash_t tag = dmScript::CheckHashOrString(L, -1);
+            AddPredicateTag(*p_predicate, tag);
```

Comment:
Do we want to know if it fails?

Reply:
Yes, maybe. Previously it also silently ignored tags above 64.

Reply:
I thought of returning a result code/boolean and log a warning.